objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/neuralNet/YoloProcessor.java

```diff
@@ -315,6 +389,45 @@ public NeuralNetResults processImage(byte[] image, Map<String, ?> request) throw
         results.setResults(foundObjects);
         return results;
     }
+    
+    /**
+     * A helper method used to crop images before they are run through the YOLO Processor, if specified in the
+     * source configuration or as query parameters. Returns the original image if an exception was caught while
+     * cropping.
+     * @param image     The byte array representation of the captured image
+     * @param x         The top left x coordinate
+     * @param y         The top left y coordinate
+     * @param w         The width of the "sub image"
+     * @param h         The height of the "sub image"
+     * @return          The cropped image resulting from the parameters above
+     */
+    public byte[] cropImage(byte[] image, int x, int y, int w, int h) {
+        try {
+            // Convert byte[] to BufferedImage and crop
+            ByteArrayInputStream bais = new ByteArrayInputStream(image);
+            BufferedImage buffImage = ImageIO.read(bais);
+            BufferedImage croppedImage = buffImage.getSubimage(x, y, w, h);
```

Comment:
do we know why we choose this mechanism (`BufferedImage`-based) instead of `OpenCV`-based?

I don't know which is right (if either), just wondering why this choice -- just because other things in YOLO are `AWT`-based?  Probably the correct choice, just wondering.